aten/src/ATen/native/mkldnn/Linear.cpp

```diff
@@ -24,22 +24,45 @@ Tensor mkldnn_linear(
 namespace at {
 namespace native {
 
+// Helper function for getting an ideep tensor out of an aten Tensor.
+// Note in case the aten Tensor is a dense tensor, make sure:
+//   1. the aten tensor is 2d, so that 3d*2d Linear is viewed as 2d*2d.
+//   2. the aten tensor is contiguous.
+inline ideep::tensor get_mkldnn_tensor(const at::Tensor& tensor) {
+  if (tensor.is_mkldnn()) {
+    return at::native::itensor_from_mkldnn(tensor);
+  } else {
+    auto x = tensor.contiguous().view({-1, tensor.size(tensor.dim() - 1)});
+    return at::native::itensor_view_from_dense(x);
+  }
+}
+
 Tensor mkldnn_linear(
     const Tensor& self,
     const Tensor& weight,
     const Tensor& bias) {
-  ideep::tensor& x = itensor_from_mkldnn(self);
-  ideep::tensor& w = itensor_from_mkldnn(weight);
+  const ideep::tensor x = get_mkldnn_tensor(self);
```

Comment:
you probably want to add TORCH_CHECK here that input ranks are >= and sizes line up

Reply:
update in `66a0ffcb`